File: ABC256/B.cpp
#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int N;
    cin >> N;
    vector<int> A(N);
    for (auto &x : A)
        cin >> x;

    vector<bool> field(4, 0);
    int point = 0;

    for (auto x : A)
    {
        vector<bool> next_field(4, 0);
        field[0] = 1;
        for (int i = 0; i < 4; i++)
        {
            if (field[i])
            {
                if (i + x >= 4)
                    point++;
                else
                    next_field[i + x] = 1;
            }
        }
        field = next_field;
    }

    cout << point << endl;
    return 0;
}

File: ABC256/B.py
n = int(input())
A = list(map(int, input().split()))
p = 0

# masu = [0 for _ in range(n)]
# for i in range(n):
#     for j in range(i + 1):
#         masu[j] += A[i]

# p = 0
# for i in range(n):
#     if masu[i] > 4:
#         p += 1
# # print(masu)
# print(p)

field = [0, 0, 0, 0]
for x in A:
    next_field = [0, 0, 0, 0]
    field[0] = 1
    for i in range(4):
        if field[i] == 1:
            if i + x >= 4:
                p += 1
            else:
                next_field[i + x] = 1
    field = next_field
print(p)